(Feature) Beautify Menu/Settings

fernandomg opened the pull request:
<!-- reporting stories from Pivotal Tracker (replace 'x' by the Story's id) -->
This PR closes [#166905175](https://www.pivotaltracker.com/story/show/166905175), by:

<!-- If you're linking a repository issue, use the following line instead:
This PR closes #x, by:
-->

* Fixing SideMenu erratic behavior
  * also disabled draggable gesture to open/close it (it may conflict with the event feed list)
* Modifying side menu to match wireframes
* Modifying Recover Wallet screen to match wireframes
* Modifying Mnemonics screen to match wireframes
* Modifying Profile Privacy to match wireframes

**Extra**:
* **Pending** to implement icons. Waiting for PR#244 to be merged, if is merged before this PR I'll update icons in this same PR. If not, will create a derived task to add them later.
 

<!-- **Considerations:** -->
<!-- * PR Title Convention: (Feature|Bug|Chore) Task Name -->
<!-- * Be as descriptive as possible, assisting reviewers as much as possible. -->
<!-- * If frontend, paste screenshots that display the UI changes. -->
<!-- * If there is interactivity, paste a gif showing the feature. -->

sirpy commented:
@fernandomg 
Is this something that happens in production also?

fernandomg commented:
> @fernandomg
> Is this something that happens in production also?

yes, that's something that was there since we have side menu.

gabitoesmiapodo commented:
More changes / suggestions for the menu:

- It looks like the contents should be covered by some kind of overlay when the menu is visible. Is this possible to do easily? I can think of a few ways to do this, but it might get difficult and it may be worth it to create another task for this.
- Looking at the design, I see the top left / bottom left borders should be rounded.
- Lastly, I think the menu should have an outer shadow.

gabitoesmiapodo commented:
About _Recover Wallet_

This part doesn't seem to match the design.

**App:**

![Screen Shot 2019-07-12 at 10 32 55](https://user-images.githubusercontent.com/4015436/61131762-706dca00-a490-11e9-905c-e77557145b15.png)

**Design:**

![Screen Shot 2019-07-12 at 10 31 56](https://user-images.githubusercontent.com/4015436/61131785-7cf22280-a490-11e9-89b4-1acbb96ee7da.png)

The rest is fine.

gabitoesmiapodo commented:
About _"Profile Privacy"_

I'm getting this (iPhone 5 emulation):

![Screen Shot 2019-07-12 at 10 40 03](https://user-images.githubusercontent.com/4015436/61132433-bb3c1180-a491-11e9-8093-6b79a07b5af6.png)

And this is the current design:

![Screen Shot 2019-07-12 at 10 40 13](https://user-images.githubusercontent.com/4015436/61132456-c727d380-a491-11e9-9894-dc292e9f871d.png)

- `Private / Masked / Public` color seems to be off, it should be `#555`
- `Private / Masked / Public` seem to overlap on narrow screens, maybe add a min-width there?
- Item's borders width should be narrower (1px I think).
- The borders should reach the screen's limits.
- `Phone Number / Email` font weight should be 500 and / or use the Roboto Medium font family.
- The `i` rounded icon at the top doesn't look like in the design (color and icon). Feel free to ignore this if it's because you're waiting for some other feature to be merged.

kbardi commented:
On focus in an input on Recovery wallet shows outline 
![image](https://user-images.githubusercontent.com/4523375/61129655-38b05380-a48b-11e9-93d9-9b6ae0660c31.png)

kbardi commented:
On Small devices like iPhone4 the side menu is broken
![image](https://user-images.githubusercontent.com/4523375/61136096-58e70f00-a499-11e9-9778-392ca371fac0.png)

kbardi commented:
Backup wallet it's also broken on smal devices, I cannot see last inputs, and inputs are not centered
![image](https://user-images.githubusercontent.com/4523375/61136169-7a47fb00-a499-11e9-8376-942226befe30.png)

kbardi commented:
Profile privacy columns are overlapped 
![image](https://user-images.githubusercontent.com/4523375/61136271-b24f3e00-a499-11e9-94b8-ed94b7c23081.png)

kbardi commented:
Ok button on tips modal looks different to the mockups
![image](https://user-images.githubusercontent.com/4523375/61136687-6b157d00-a49a-11e9-994a-9d571e8121d5.png)

Mockup:
![image](https://user-images.githubusercontent.com/4523375/61136750-854f5b00-a49a-11e9-9839-369cd43355ab.png)

fernandomg commented:
@kbardi, @gabitoesmiapodo 

### SideMenu
- changed sideMenu `fontFamily` to medium
- added paddingRight of `16`
- changed paddingTop/Bottom from `14` to `16` to match the "multiple of 4" rule
- removed `openMenuOffset` restriction of `240` (this was an intention to match wireframes values), to let the library calculate the proper size of the menu.
- also, set text alignment to `left`, so it looks aligned when menu option goes multi-line
- updated icons based on wireframes
- **NOTE:** all about blurring/shadows...
  - tried to use `react-navigation-drawer`, it doesn't work well on web (it has dependencies that weren't fully migrated to web). See: [#167276585](https://www.pivotaltracker.com/story/show/167276585).
- **NOTE:** options on side menu don't match with wireframes. But there's no further information to where `FAQ` and `About` must lead to.
  - Left a question to Liav in PT.
- **NOTE:** regarding icons, there's no icon for `Privacy Policy` (I used the `faq` one). Also, the option is not in the wireframes.
  - Left a question to Liav in PT.

### ProfilePrivacy
- changed fonts color to `#555` (added to theme.colors as `gray`)
- properly set fonts to 'Regular Medium' where required
- fixed lateral margins
- fixed overlapping of columns titles
- changed (i) icon color to `primary` and update its icon
- changed color of inactive radio buttons to `gray`
- changed row lines width to `normalize(1)`, not using `StyleSheet.hairlineWidth` as it's not being visible
- **NOTE:** navigation from `topNavBar` is not working properly. If user access profile page from the side menu, tapping back arrow will lead him to the Profile screen. See: [#167276594](https://www.pivotaltracker.com/story/show/167276594)
- **NOTE:** Despite I'm not seeing any longer the "Tips" dialog in the wireframes. I didn't make a huge effort in its stylings as I assumed it to be part of the modals/dialogs beautify task.


### Mnemonics
- Changed "Resend backup email" from `Text` to `CustomButton`.
- Unwrapped theme's values of `normalize` (those are already normalized)
- Set to `overflowY=auto` for the input fields container, to handle small devices.
- **NOTE:** tried to use `ScrollView` but experienced issues with elements disposition, so I went for the `overflowY` solution.
